metrics-operator/api/v1alpha3/analysisdefinition_webhook.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+// log is for logging in this package.
+var analysisdefinitionlog = logf.Log.WithName("analysisdefinition-webhook")
+
+func (r *AnalysisDefinition) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+//+kubebuilder:webhook:path=/validate-metrics-keptn-sh-v1alpha3-analysisdefinition,mutating=false,failurePolicy=fail,sideEffects=None,groups=metrics.keptn.sh,resources=analysisdefinitions,verbs=create;update,versions=v1alpha3,name=vanalysisdefinition.kb.io,admissionReviewVersions=v1
+
+var _ webhook.Validator = &AnalysisDefinition{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *AnalysisDefinition) ValidateCreate() error {
+	analysisdefinitionlog.Info("validate create", "name", r.Name)
+
+	for _, o := range r.Spec.Objectives {
+		if err := o.validate(); err != nil {
+			return err
+		}
+	}
+
+	return r.Spec.TotalScore.validate()
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *AnalysisDefinition) ValidateUpdate(old runtime.Object) error {
+	analysisdefinitionlog.Info("validate update", "name", r.Name)
+
+	for _, o := range r.Spec.Objectives {
+		if err := o.validate(); err != nil {
+			return err
+		}
+	}
+
+	return r.Spec.TotalScore.validate()
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *AnalysisDefinition) ValidateDelete() error {
```

Comment:
do we need this? in the kubebuilder comment (L37) only create/update are registered

Reply:
it can be removed surely, but was auto-generated so I left it there